src/db/migrations/20230907143204_add_onerep_scanresults_table.js

```diff
@@ -0,0 +1,87 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+export async function up(knex) {
+  await knex.schema.alterTable("onerep_scans", table => {
+    table.unique("onerep_scan_id");
+  });
+
+  await knex.schema
+    .createTable("onerep_scan_results", table => {
+      table.increments("id").primary();
+      table.integer("onerep_scan_result_id").notNullable();
+      table.integer("onerep_scan_id").references("onerep_scans.onerep_scan_id").notNullable();
```

Comment:
One thing I'm not storing here is the `onerep_profile_id`, since that can already be obtained by joining with `onerep_scans`, so this saves us an extra column. I will have to do that join (and then another join to the `subscribers` table) in my follow-up PR. I'm not really sure on the trade-offs here, i.e. whether an extra join in a query outside of the hot path is preferable compared to having a redundant extra integer column.

Reply:
Choosing not to store is the right decision here. This is a classic tradeoff between normalization / performance and joins definitely have a bad rep. However, we should always start with the simplest approach and stick with the core principles of relational database (ACID), and iterate if necessary ONLY when we have metrics that support performance bottlenecks. By duplicating data, we essentially nullifies consistency, which is the C there.